### PR TITLE
Enforce log.WithError(err) static analysis and fix all violations

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

Enables the static analysis from #11143 and fixes all violations.

**Which issues(s) does this PR fix?**

**Other notes for review**

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Oops, fixing tests...

File: beacon-chain/sync/pending_blocks_queue.go [hunk 1]
```diff
@@ -167,7 +167,7 @@ func (s *Service) processPendingBlocks(ctx context.Context) error {
 						s.setBadBlock(ctx, blkRoot)
 					}
 				}
-				log.Debugf("Could not process block from slot %d: %v", b.Block().Slot(), err)
+				log.WithError(err).WithField("slot", b.Block().Slot()).Debug("Could not process block from slot")
```
Comment (potuz) [inline beacon-chain/sync/pending_blocks_queue.go:170]:
```suggestion
				log.WithError(err).WithField("slot", b.Block().Slot()).Debug("Could not process block")
```

File: tools/exploredb/main.go [hunk 2]
```diff
@@ -104,7 +104,7 @@ func main() {
 	case "migration-check":
 		destDbNameWithPath := filepath.Join(*destDatadir, *dbName)
 		if _, err := os.Stat(destDbNameWithPath); os.IsNotExist(err) {
-			log.Fatalf("could not locate destination database file : %s, %v", destDbNameWithPath, err)
+			log.WithError(err).WithField("path", dbNameWithPath).Fatal("could not locate database file")
```
Comment (potuz) [inline tools/exploredb/main.go:107]: why the change here instead of using `destDbNameWithPath`?

File: tools/exploredb/main.go [hunk 3]
```diff
@@ -414,11 +414,11 @@ func checkValidatorMigration(dbNameWithPath, destDbNameWithPath string) {
 	for rowCount, key := range sourceStateKeys[910:] {
 		sourceState, stateErr := sourceDB.State(ctx, bytesutil.ToBytes32(key))
 		if stateErr != nil {
-			log.Fatalf("could not get from source db, the state for key : %s, %v", hexutils.BytesToHex(key), stateErr)
+			log.WithError(stateErr).WithField("key", hexutils.BytesToHex(key)).Fatalf("could not get from source db, the state for key")
 		}
 		destinationState, stateErr := destDB.State(ctx, bytesutil.ToBytes32(key))
 		if stateErr != nil {
-			log.Fatalf("could not get destination db, the state for key : %s, %v", hexutils.BytesToHex(key), stateErr)
+			log.WithError(stateErr).WithField("key", hexutils.BytesToHex(key)).Fatalf("could not get from source db, the state for key")
```
Comment (potuz) [inline tools/exploredb/main.go:421]: is it source or destination?

File: tools/exploredb/main.go (hunk 2 above)
Comment (prestonvanloon) [reply tools/exploredb/main.go:107]: oh... copy paste error!

File: tools/exploredb/main.go (hunk 3 above)
Comment (prestonvanloon) [reply tools/exploredb/main.go:421]: Another copy paste fail, thanks!

**Comment by rkapka** [comment]
Thank you for contributing to the repo! Please sign the CLA before we can proceed :laughing:

**Review by potuz — APPROVED** [review]
